src/App.tsx

```diff
@@ -1,9 +1,64 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { todosSlice } from './features/todos';
+import { Status } from './types/Status';
 
-export const App = () => (
-  <>
+export const App = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status) as Status;
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const loadData = async () => {
+      try {
+        const data = await getTodos();
+
+        dispatch(todosSlice.actions.setTodos(data));
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadData();
+  }, [dispatch]);
```

Comment:
The dependency array for `useEffect` currently only includes `dispatch`. Ensure that all variables used within the effect that are not constants or derived from `dispatch` are included in the dependency array to avoid potential stale closures.